direct/common/subsample.py

```diff
@@ -243,6 +246,83 @@ def mask_func(
+    """
+
+    def __init__(
+        self,
+        accelerations: Union[List[Number], Tuple[Number, ...]],
+        center_fractions: Optional[Union[List[int], Tuple[int, ...]]] = None,
+        uniform_range: bool = False,
+    ):
+        """Inits :class:`CartesianRandomMaskFunc`.
+
+        Parameters
+        ----------
+        accelerations: Union[List[Number], Tuple[Number, ...]]
+            Amount of under-sampling_mask. An acceleration of 4 retains 25% of the k-space, the method is given by
+            mask_type. Has to be the same length as center_fractions if uniform_range is not True.
+        center_fractions: list or tuple of ints, optional
+            Number of low-frequency (center) columns to be retained.
+            If multiple values are provided, then one of these numbers is chosen uniformly each time.
+        uniform_range: bool
+            If True then an acceleration will be uniformly sampled between the two values. Default: True.
+        """
+        super().__init__(
+            accelerations=accelerations,
+            center_fractions=center_fractions,
+            uniform_range=uniform_range,
+        )
+
+    def mask_func(
+        self,
+        shape: Union[List[int], Tuple[int, ...]],
+        return_acs: bool = False,
+        seed: Optional[Union[int, Iterable[int]]] = None,
+    ) -> torch.Tensor:
+        """Creates a random vertical Cartesian mask.
+
+        Parameters
+        ----------
+        shape: list or tuple of ints
+            The shape of the mask to be created. The shape should at least 3 dimensions.
+            Samples are drawn along the second last dimension.
+        return_acs: bool
+            Return the autocalibration signal region as a mask.
+        seed: int or iterable of ints or None (optional)
+            Seed for the random number generator. Setting the seed ensures the same mask is generated
+             each time for the same shape. Default: None.
+
+
+        Returns
+        -------
+        mask: torch.Tensor
+            The sampling mask.
+        """
+        if len(shape) < 3:
+            raise ValueError("Shape should have 3 or more dimensions")
+
+        with temp_seed(self.rng, seed):
+            num_cols = shape[-2]
+
+            num_center_lines, acceleration = self.choose_acceleration()
+
+            mask = self.center_mask_func(num_cols, num_center_lines)
+
+            if return_acs:
+                return torch.from_numpy(self._reshape_and_broadcast_mask(shape, mask))
+
+            # Create the mask
+            prob = (num_cols / acceleration - num_center_lines) / (num_cols - num_center_lines)
+            mask = mask | (self.rng.uniform(size=num_cols) < prob)
+
+        return torch.from_numpy(self._reshape_and_broadcast_mask(shape, mask))
+
+
 class FastMRIEquispacedMaskFunc(FastMRIMaskFunc):
     r"""Equispaced vertical line mask function.
 
```

Comment:
Same comment

Reply:
Both Cartesian and FastMRI classes inherit from a single class now